Content.Client/_DV/AACTablet/UI/AACWindow.xaml.cs

```diff
@@ -27,15 +28,29 @@ public AACWindow()
     {
         RobustXamlLoader.Load(this);
         IoCManager.InjectDependencies(this);
+        _phrases = _prototype.EnumeratePrototypes<QuickPhrasePrototype>().ToList();
+        SearchBar.OnTextChanged += FilterSearch;
         PopulateGui();
+        FilterSearch(null);
     }
 
-    private void PopulateGui()
+    private void FilterSearch(LineEdit.LineEditEventArgs? obj)
     {
-        var phrases = _prototype.EnumeratePrototypes<QuickPhrasePrototype>().ToList();
+        SearchResults.DisposeAllChildren();
+        var emptySearch = string.IsNullOrEmpty(SearchBar.Text);
+        var searchedPhrases = _phrases
+            .Where(it => emptySearch || Loc.GetString(it.Text).StartsWith(SearchBar.Text, StringComparison.CurrentCultureIgnoreCase))
+            .GroupBy(it => it.Group)
+            .OrderBy(it => it.Key)
+            .ToDictionary(group => group.Key, group => group.OrderBy(it => Loc.GetString(it.Text)).ToList());
```

Comment:
this is way too much linq create 1 dictionary and do it manually
also searching should use contains not start with

Reply:
I personally found the search results felt better with starts with than contains, but I'm fine changing it if contains is the preferred option.

Can you please elaborate on the one dictionary manually?

Reply:
using linq especially chaining multiple calls like this, and even nesting them, does a huge number of unnecessary allocations
"unroll" it and do it by hand by
1. creating a single dictionary for the control in its constructor
2. in this, `Clear`ing it then iterate each phrase and add it to an existing or new group list with the filtering logic
3. at the end iterate the dictionary and sort each value's list

copy whatever criminal records uses for searching though, people are used to it

Reply:
Understood, wasn't the most familiar with Linq. Will fix